### ISSUE TITLE
docker, docker-compose and nginx stack: unable to run tasks

### ISSUE THREAD

**Issue by Destrocamil** [ansible-semaphore/semaphore#1485]
Hello,
I have a docker-compose file with nginx, semaphore and the DB.
If I run this stack locally, everything works.
However, if I run this stack on a remote machine (aka monitoring-local), I can set up Semaphore, add tasks, templates (...) without any issue, but when I run a task, it just hangs.
Parallel is set to 0, all the required tokens and keys are there but it does not work.
Web UI logs
```
Waiting state
 11:47:03 AM
Task 10 added to queue

```
Semaphore logs
```
Attempting to connect to database semaphore on postgres:5432 with user semaphore ...
Postgres semaphore@postgres:5432 semaphore
Tmp Path (projects home) /tmp/semaphore
Semaphore v2.9.11
Interface 
Port :3000
Server is running
time="2023-09-13T13:44:52Z" level=info msg="Task 4 added to queue"
time="2023-09-13T13:44:53Z" level=info msg="Set resource locker with TaskRunner 4"
time="2023-09-13T13:44:53Z" level=info msg="Task 4 removed from queue"
time="2023-09-13T13:44:57Z" level=info msg="Stopped running TaskRunner 4"
time="2023-09-13T13:44:57Z" level=info msg="Release resource locker with TaskRunner 4"
time="2023-09-13T15:27:15Z" level=info msg="Task 5 added to queue"
time="2023-09-13T15:27:18Z" level=info msg="Set resource locker with TaskRunner 5"
time="2023-09-13T15:27:18Z" level=info msg="Task 5 removed from queue"
time="2023-09-13T15:27:23Z" level=info msg="Stopped running TaskRunner 5"
time="2023-09-13T15:27:23Z" level=info msg="Release resource locker with TaskRunner 5"
time="2023-09-13T15:38:23Z" level=info msg="Task 6 added to queue"
time="2023-09-13T15:38:28Z" level=info msg="Set resource locker with TaskRunner 6"
time="2023-09-13T15:38:28Z" level=info msg="Task 6 removed from queue"
time="2023-09-13T15:38:32Z" level=warning msg="read |0: file already closed" error="Failed to read TaskRunner output"
time="2023-09-13T15:38:32Z" level=warning msg="read |0: file already closed" error="Failed to read TaskRunner output"
time="2023-09-13T15:38:32Z" level=info msg="Stopped running TaskRunner 6"
time="2023-09-13T15:38:32Z" level=info msg="Release resource locker with TaskRunner 6"
time="2023-09-13T15:46:55Z" level=info msg="Task 7 added to queue"
time="2023-09-13T15:46:58Z" level=info msg="Set resource locker with TaskRunner 7"
time="2023-09-13T15:46:58Z" level=info msg="Task 7 removed from queue"
time="2023-09-13T15:47:04Z" level=info msg="Stopped running TaskRunner 7"
time="2023-09-13T15:47:04Z" level=info msg="Release resource locker with TaskRunner 7"
time="2023-09-14T09:17:12Z" level=info msg="Task 8 added to queue"
time="2023-09-14T09:17:13Z" level=info msg="Set resource locker with TaskRunner 8"
time="2023-09-14T09:17:13Z" level=info msg="Task 8 removed from queue"
time="2023-09-14T09:17:26Z" level=info msg="Stopped running TaskRunner 8"
time="2023-09-14T09:17:26Z" level=info msg="Release resource locker with TaskRunner 8"
time="2023-09-14T09:22:43Z" level=info msg="Task 9 added to queue"
time="2023-09-14T09:22:48Z" level=info msg="Set resource locker with TaskRunner 9"
time="2023-09-14T09:22:48Z" level=info msg="Task 9 removed from queue"
time="2023-09-14T09:22:48Z" level=warning msg="read |0: file already closed" error="Failed to read TaskRunner output"
time="2023-09-14T09:23:01Z" level=info msg="Stopped running TaskRunner 9"
time="2023-09-14T09:23:01Z" level=info msg="Release resource locker with TaskRunner 9"
time="2023-09-14T09:47:03Z" level=info msg="Task 10 added to queue"
time="2023-09-14T09:47:08Z" level=info msg="Set resource locker with TaskRunner 10"
time="2023-09-14T09:47:08Z" level=info msg="Task 10 removed from queue"
time="2023-09-14T09:47:22Z" level=info msg="Stopped running TaskRunner 10"
time="2023-09-14T09:47:22Z" level=info msg="Release resource locker with TaskRunner 10"
```

Any idea? 
Thanks!

EDIT: exposing via port 3000, bypassing nginx, it works

**Comment by kenavn** [comment]
Hi,
I experienced the same thing yesterday and today, except that I now see this both with and without a proxy. I've tested this in two different environments (on docker) and the tasks do not run.

**It seems to be a problem with the latest docker build (v2.9.21)**

This is the image definition as specified in the docs (a line from the docker-compose.yml):
`image: semaphoreui/semaphore:latest`

This is an image I have tested with that actually works;
`image: semaphoreui/semaphore:v2.9.18`

I guess the docs use the latest-tag to be universally valid, although setting a specific version is preferred if you don't want surprises like this..

**Comment by Yuri-Lima** [comment]
Yes, @Destrocamil i've actually just update to that latest image `v2.9.21` and i got the same output result as yours.

However, i could see the image @kenavn tested `v2.9.18` is working fine indeed. Thank you fot that 👍

The version `v2.9.21` is comming from [semaphoreui/semaphore/tags](https://hub.docker.com/r/semaphoreui/semaphore/tags)

**Comment by fiftin** [comment]
Fixing in progress ;)

**Comment by Destrocamil** [comment]
Thank you @all. 
Seems it's really th version. I've set it to 2.9.18 and it's working :)

**Comment by stiltzkin10** [comment]
Has this been fixed? I'm testing 2.10.22 and the output is stuck when using nginx as a proxy.

Without nginx the output works fine.
